tests/components/squeezebox/test_config_flow.py

```diff
@@ -0,0 +1,263 @@
+"""Test the Logitech Squeezebox config flow."""
+from asynctest import patch
+from pysqueezebox import Server
+
+from homeassistant import config_entries
+from homeassistant.components.squeezebox.const import DOMAIN
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    HTTP_UNAUTHORIZED,
+)
+from homeassistant.data_entry_flow import (
+    RESULT_TYPE_ABORT,
+    RESULT_TYPE_CREATE_ENTRY,
+    RESULT_TYPE_FORM,
+)
+
+from tests.common import MockConfigEntry
+
+HOST = "1.1.1.1"
+HOST2 = "2.2.2.2"
+PORT = 9000
+UUID = "test-uuid"
+UNKNOWN_ERROR = "1234"
+
+
+async def mock_discover(_discovery_callback):
+    """Mock discovering a Logitech Media Server."""
+    _discovery_callback(Server(None, HOST, PORT, uuid=UUID))
+
+
+async def mock_failed_discover(_discovery_callback):
+    """Mock unsuccessful discovery by doing nothing."""
+
+
+async def patch_async_query_unauthorized(self, *args):
+    """Mock an unauthorized query."""
+    self.http_status = HTTP_UNAUTHORIZED
+    return False
+
+
+async def test_user_form(hass):
+    """Test user-initiated flow, including discovery and the edit step."""
+    with patch("pysqueezebox.Server.async_query", return_value={"uuid": UUID},), patch(
+        "homeassistant.components.squeezebox.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.squeezebox.async_setup_entry", return_value=True,
+    ) as mock_setup_entry, patch(
+        "homeassistant.components.squeezebox.config_flow.async_discover", mock_discover
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "edit"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5737505)